src/entries/popup/pages/home/Activity/useTransaction.ts

```diff
@@ -90,3 +106,107 @@ export function useTransaction({
       queryClient.getQueryState(paginatedTransactionsKey)?.dataUpdatedAt,
   });
 }
+
+const getCustomChainTransaction = async ({
+  chainId,
+  hash,
+}: {
+  chainId: number;
+  hash: Hash;
+}) => {
+  const provider = getProvider({ chainId });
+  const transaction = await provider.getTransaction(hash);
+  if (!transaction) return undefined;
+
+  const block = transaction?.blockHash
+    ? await provider.getBlock(transaction?.blockHash)
+    : undefined;
+
+  const parsedTransaction = transaction.blockNumber
+    ? ({
+        status: 'confirmed',
+        blockNumber: transaction.blockNumber || 0,
+        minedAt: block?.timestamp || 0,
+        confirmations: transaction.confirmations,
+        gasUsed: transaction.gasLimit.toString(),
+        hash: transaction.hash as `0x${string}`,
+        nonce: transaction.nonce,
+        chainId: transaction.chainId,
+        from: transaction.from as `0x${string}`,
+        to: transaction.to as `0x${string}`,
+        data: transaction.data,
+        value: transaction.value.toString(),
+        type: 'send',
+        title: i18n.t('transactions.send.confirmed'),
+        baseFee: block?.baseFeePerGas?.toString(),
+        maxFeePerGas: transaction.maxFeePerGas?.toString(),
+        maxPriorityFeePerGas: transaction.maxPriorityFeePerGas?.toString(),
+        gasPrice: transaction.gasPrice?.toString(),
+      } satisfies MinedTransaction)
+    : ({
+        status: 'pending',
+        hash: transaction.hash as `0x${string}`,
+        nonce: transaction.nonce,
+        chainId: transaction.chainId,
+        from: transaction.from as `0x${string}`,
+        to: transaction.to as `0x${string}`,
```

Comment:
At some point we should create a `TxHash` type

Reply:
i started using the Hash from wagmi in some places